src/parser/parser_support.c

```diff
@@ -10968,39 +10968,198 @@ pt_get_server_name_list (PARSER_CONTEXT * parser, PT_NODE * node, void *arg, int
   return node;
 }
 
+static char *
+skip_comment (char *ps)
+{
+  while (*ps)
+    {
+      if (ps[0] == '*' && ps[1] == '/')
+	{
+	  ps += 2;
+	  break;
+	}
+      ps++;
+    }
+  return ps;
+}
+
+char *
+read_id (char *ps, char end_char)
+{
+  while (*ps != end_char)
+    {
+      ps++;
+    }
+
+  return ((*ps == end_char) ? (ps + 1) : ps);
+
+}
+
+static char *
+read_string (char *ps, char end_char, bool no_escape)
+{
+  if (no_escape)
+    {
+      while (*ps)
+	{
+	  if (ps[0] == end_char)
+	    {
+	      if (ps[1] != end_char)
+		{
+		  return ps + 1;
+		}
+	      ps++;
+	    }
+	  ps++;
+	}
+    }
+  else
+    {
+      while (*ps)
+	{
+	  if (ps[0] == end_char)
+	    {
+	      if (ps[1] != end_char)
+		{
+		  return ps + 1;
+		}
+	      ps++;
+	    }
+
+	  else if (ps[0] == '\\')
+	    {
+	      switch (ps[1])
+		{
+		case '\'':
+		case '"':
+		case 'n':
+		case 'r':
+		case 't':
+		case '\\':
+		case '%':
+		case '_':
+		  ps++;
+		  break;
+
+		default:
+		  break;
+		}
+	    }
+	  ps++;
+	}
+    }
+
+  return ps;
+}
+
+static char *
+find_circle_at_char (bool ansi_quotes, bool no_escape, char *ps)
+{
+  while (*ps)
+    {
+      switch (*ps)
+	{
+	case '@':
+	  {
+	    return ps;
+	  }
+	  break;
+
+	case '[':
+	  ps = read_id (ps + 1, ']');
+	  break;
+
+	case '`':
+	  ps = read_id (ps + 1, '`');
+	  break;
+
+	case '"':
+	  ps = (ansi_quotes ? read_id (ps + 1, '"') : read_string (ps + 1, '"', no_escape));
+	  break;
+
+	case '\'':
+	  ps = read_string (ps + 1, '\'', no_escape);
+	  break;
+
+	case '/':
+	case '-':
+	  if (ps[0] == ps[1])
+	    {
+	      ps += 2;
+	      while (*ps)
+		{
+		  ps++;
+		  if (ps[-1] == '\n')
+		    {
+		      break;
+		    }
+		}
+	    }
+	  else if (ps[0] == '/' && ps[1] == '*')
+	    {
+	      ps = skip_comment (ps + 2);
+	    }
+	  break;
+
+	default:
+	  ps++;
+	  break;
+	}
+    }
+
+  return NULL;
+}
+
 static PARSER_VARCHAR *
 pt_make_remote_query (PARSER_CONTEXT * parser, char *sql_user_text, SERVER_NAME_LIST * snl)
 {
   PARSER_VARCHAR *pvc = NULL;
   char *ps, *pt, *t;
+  int adjust_pos = 1;
+  bool ansi_quotes = prm_get_bool_value (PRM_ID_ANSI_QUOTES);
+  bool no_escape = prm_get_bool_value (PRM_ID_NO_BACKSLASH_ESCAPES);
```

Comment:
I think it is more efficient to copy `PT_NODE *` here, process the necessary parts, and then use the `parser_print_tree` function to make `char * sql_user_text`.

Reply:
This function removes the @server information from the query entered by the user and creates a query string to be transmitted to the remote server.

The reason for using sql_user_text is as follows.

Here we need the user's original query that has not been rewritten in the middle.
It is not easy to get the top root node of the tree.

Reply:
This function will not be used, because I change the routine to use parser_print_tree or pt_print_bytes to add the custom print flag "PT_SUPPRESS_SERVER_NAME".